app/controllers/discovered_hosts_controller.rb

```diff
@@ -80,6 +80,7 @@ def update
           delete_discovery_attribute_set(@host.id)
           process_success :success_redirect => host_path(@host), :redirect_xhr => request.xhr?
         else
+          @host.expire_token
```

Comment:
Maybe we missed a case (UI/CLI) when an explicit transaction is not created. We now explicitly check them, therefore it is not possible to change to discovered host without a transaction.